### PR TITLE
Add two options to include and exclude files on upload

### PR DESCRIPTION
**Pull request by pacoorozco**
This PR is trying to solve #44 by adding two new configuration options `includePatterns` and `excludePatterns` to allow users to white list and black list files to be uploaded.

@avidrissman  @rfgamaral @tonymet @nhorvath @nmrshll

### CODE REVIEW

**Comment by rfgamaral** [comment]
Sorry @pacoorozco, Go is not my strong suite so I don't think I'd be much help reviewing this.

**Comment by pacoorozco** [comment]
It's ready to be merged, can you review it?

@avidrissman @tonymet @nhorvath @nmrshll

File: .docs/configuration.md [hunk 1]
```diff
@@ -79,4 +81,23 @@ If set to true, media will be deleted from local disk after upload.
...
+
+For example, to upload all _JPG and PNG files_ that are not named _*ScreenShots*_ you can configure it like this:
+```
+      includePatterns: [ "*.jpg", "*.png" ]
+      excludePatterns: [ "*ScreenShot*" ]
+```
+
+Patterns use [filepath.Glob](https://golang.org/pkg/path/filepath/#Glob) internally, see [filepath.Match](https://golang.org/pkg/path/filepath/#Match) for syntax. 
+
+Regular wildcards cannot be used to match over the directory separator `/`. For example: `b*ash` matches `/dir/bash` but does not match `/dir/ash`.
```
Comment (avidrissman) [inline .docs/configuration.md]: Why would people think that `b*ash` might match `/dir/ash`? Perhaps:

> Regular wildcards cannot be used to match over the directory separator `/`. For example: `b*ash` matches `/dir/bash` but does not match `/dirb/ash`.

Comment (pacoorozco) [reply .docs/configuration.md]: You're right, there is a type... Thanks

**Pull request merged**